packages/office-ui-fabric-react/src/components/SwatchColorPicker/SwatchColorPicker.tsx

```diff
@@ -0,0 +1,580 @@
+  /**
+   * Render the swatch color picker with a button in the given (collapased vs expanded state)
+   * @returns {JSX.Element[]} - an array of the elements to render
+   */
+  private _buttonToRender(): JSX.Element[] {
+    let { selectedIndex, expanded } = this.state;
+    let {
+      swatchColorPickerButtonProps,
+      swatchColorPickerItems,
+      updateButtonIconWithColor,
+      width,
+      disabled
+    } = this.props;
+
+    let colorToSet = null;
+
+    // Do we need to update the button with the selected
+    // item's color? If so, attempt to grab the color
+    if (updateButtonIconWithColor &&
+      selectedIndex &&
+      selectedIndex >= 0 &&
+      selectedIndex < swatchColorPickerItems.length) {
+      let swatchColorPickerItem = swatchColorPickerItems[selectedIndex];
+      if (swatchColorPickerItem.type === SwatchColorPickerItemType.Cell &&
+        swatchColorPickerItem.color !== null &&
+        swatchColorPickerItem.color.length > 0) {
+        colorToSet = swatchColorPickerItem.color;
+      }
+    }
+
+    let elements: JSX.Element[] = [];
+
+    elements.push(<DefaultButton
+      { ...swatchColorPickerButtonProps }
+      style={ { color: colorToSet && colorToSet } }
+      className={
+        css('ms-swatchColorPickerButton',
+          {
+            'is-expanded': expanded
+          }
+        )
+      }
+      onClick={ this._onButtonClick }
+      aria-haspopup={ true }
+      aria-expanded={ (!this.props.disabled && this.state.expanded) ? true : false }
+      disabled={ this.props.disabled }
+      menuIconProps={ !swatchColorPickerButtonProps.menuIconProps ? { iconName: 'chevronDown' } : swatchColorPickerButtonProps.menuIconProps }
+    />);
+
+    // Add the menu if we are endabled and expanded
+    if (!this.props.disabled && this.state.expanded) {
+      elements.push(this._onRenderContainer());
+    }
+
+    return elements;
+  }
+
+  /**
+   * Render the full swatch color picker this happens for both
+   * the always visible case as well as the expandable case (when rendered in a menu)
+   * @returns {JSX.Element} - The element representing the fully rendered swatch color picker
+   */
+  @autobind
+  private _fullSwatchColorPickerToRender(): JSX.Element {
+    let { swatchColorPickerItems, width } = this.props;
+    return (
+      <FocusZone
+        isCircularNavigation={ true }
+        className={ css('ms-swatchColorPickerBodyContainer', styles.swatchColorPickerContainer) }
+      >
+        { this._onRenderItems(swatchColorPickerItems.map((item, index) => { return { ...item, index }; })) }
+      </FocusZone>
+    );
+  }
+
+  /**
+   * Render all the items
+   * @param items - The swatch color picker items
+   * @returns {JSX.Element[]} - An array of all the items in the swatch color picker
+   */
+  @autobind
+  private _onRenderItems(items: ISwatchColorPickerItemProps[]): JSX.Element[] {
+    let {
+      width
+    } = this.props;
+
+    // Make sure we have a cell that we need to render
```

Comment:
That is fast fail to keep us from doing extra work if not needed. After thinking about this again, not passing in any cells isn't really a valid scenario so I've removed this line of code